staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -125,6 +125,9 @@ func RunServer(
 		var listener net.Listener
 		listener = tcpKeepAliveListener{ln.(*net.TCPListener)}
 		if server.TLSConfig != nil {
+			if len(server.TLSConfig.Certificates) == 0 && server.TLSConfig.GetCertificate == nil {
+				glog.Fatalf("tls: neither Certificates nor GetCertificate set in Config")
```

Comment:
what happens now with the patch in this case?

Reply:
If we startup kube-controller-manager with secure server enabled `--secure-port=xxxx`, but not provide certificate, it will exit. 
And donot have any influence on kube-apiserever.